pyproj/_crs.pyx

```diff
@@ -257,8 +257,117 @@ cdef class PrimeMeridian:
         prime_meridian.unit_name = pystrdecode(unit_name)
         return prime_meridian
 
+cdef class Datum:
+    def __cinit__(self):
+        self.projobj = NULL
+        self.projctx = NULL
+        self.name = None
+        self._ellipsoid = None
+        self._prime_meridian = None
+
+    def __dealloc__(self):
+        """destroy projection definition"""
+        if self.projobj != NULL:
+            proj_destroy(self.projobj)
+        # let the parent CRS class manage this
+        # if self.projctx != NULL:
+        #     proj_context_destroy(self.projctx)
+
+
+    @staticmethod
+    cdef create(PJ_CONTEXT* projcontext, PJ* projobj, int horizontal):
```

Comment:
Is there a reason that you use here a static method `create` (and as you do in the other Ellipsoid et al classes), while in the `_CRS` class you use a more typical `__init__` ?

Reply:
The reason is because the `__init__` does not support the PJ_CONTEXT* typing. But, the staticmethod does.